lib/improver/tests/ensemble_calibration/ensemble_calibration/test_ApplyCoefficientsForEnsembleCalibration.py

```diff
@@ -113,150 +113,211 @@ class Test__init__(IrisTest):
 
     """Test the __init__ method."""
 
-    def setUp(self):
-        """Set up test cubes."""
-        data = np.ones([2, 2], dtype=np.float32)
-        self.current_forecast = set_up_variable_cube(data)
-        coeff_names = ["gamma", "delta", "alpha", "beta"]
-        coeff_values = np.array([0, 1, 2, 3], np.int32)
-        self.coefficients_cube, _, _ = create_coefficients_cube(
-            self.current_forecast, coeff_names, coeff_values)
-
     def test_basic(self):
-        """Test without specifying any keyword arguments."""
-        plugin = Plugin(self.current_forecast, self.coefficients_cube)
-        self.assertEqual(plugin.current_forecast, self.current_forecast)
-        self.assertEqual(plugin.coefficients_cube, self.coefficients_cube)
-
-    def test_with_kwargs(self):
-        """Test without specifying any keyword arguments."""
-        plugin = Plugin(self.current_forecast, self.coefficients_cube,
-                        predictor_of_mean_flag="realizations")
-        self.assertEqual(plugin.current_forecast, self.current_forecast)
-        self.assertEqual(plugin.coefficients_cube, self.coefficients_cube)
-
-    def test_mismatching_coordinates(self):
-        """Test if there is a mismatch in the forecast_period coordinate."""
-        self.current_forecast.coord("forecast_period").convert_units("hours")
-        msg = "The forecast_period coordinate of the current forecast cube"
-        with self.assertRaisesRegex(ValueError, msg):
-            Plugin(self.current_forecast, self.coefficients_cube)
-
-    def test_matching_domain(self):
-        """Test whether the domain of the forecast and the domain of the
-        coefficients cube matches."""
-        current_forecast = self.current_forecast[0, :]
-        msg = "The domain along the"
+        """Test without specifying a predictor_of_mean_flag."""
+        plugin = Plugin()
+        self.assertEqual(plugin.predictor_of_mean_flag, "mean")
+
+    def test_with_predictor_of_mean_flag(self):
```

Comment:
I have streamlined the tests in ```Test_check_predictor_of_mean_flag```. I have removed the invalid predictor test from this init method set of tests as you are correct that it is testing the other function rather than anything inherent to this code.  I have left the test that checks using the flag actually makes a change to the internal variable of this plugin just to demonstrate that this works and that the default is not always used.